### ISSUE TITLE
Non-streaming logs coming as ascii 

### ISSUE THREAD

**Issue by eljrax** [docker/docker-py#2005]
TL;DR - when using `container.logs()` with `stream=False`, log messages comes through one byte at a time. As ASCII values in Python 3.5.2 and letters in 2.7.6.

Given this code: 
```python
from docker import DockerClient


def main():
    command = "/bin/bash -c 'echo stdout string ; (>&2 echo stderr string) ; sleep 5'"
    docker = DockerClient(base_url='unix://var/run/docker.sock')
    container = docker.containers.create("ubuntu:16.04", command=command)

    container.start()
    container.reload()
    for logline in container.logs(stdout=True, stderr=False, stream=True):
        print("stdout: {0}".format(logline))

    stderr_as_ascii = ''
    for logline in container.logs(stdout=False, stderr=True, stream=False):
        print("stderr: {0}".format(logline))
        stderr_as_ascii += chr(logline)

    print("Actual stderr: {0}".format(stderr_as_ascii))


if __name__ == "__main__":
    main()
```

The output for Python 3 is:
```
$ ./docker_test.py 
stdout: b'stdout string\n'
stderr: 115
stderr: 116
stderr: 100
stderr: 101
stderr: 114
stderr: 114
stderr: 32
stderr: 115
stderr: 116
stderr: 114
stderr: 105
stderr: 110
stderr: 103
stderr: 10
Actual stderr: stderr string
```

And for Python 2 (removed the `chr()` code as we're no longer getting integers):
```
# ./docker_test.py 
stdout: stdout string

stderr: s
stderr: t
stderr: d
stderr: e
stderr: r
stderr: r
stderr:  
stderr: s
stderr: t
stderr: r
stderr: i
stderr: n
stderr: g
stderr: 
```

If I set `stream=True` on the call to logs to get stderr as well, I get the expected behaviour:
```
# ./docker_test.py 
stdout: b'stdout string\n'
stderr: b'stderr string\n'
```

Versions:
```
# python3 --version
Python 3.5.2
# pip freeze | egrep "docker|requests"
docker==3.2.1
docker-pycreds==0.2.2
requests==2.18.4

# docker version
Client:
 Version:	18.03.0-ce
 API version:	1.37
 Go version:	go1.9.4
 Git commit:	0520e24
 Built:	Wed Mar 21 23:10:01 2018
 OS/Arch:	linux/amd64
 Experimental:	false
 Orchestrator:	swarm

Server:
 Engine:
  Version:	18.03.0-ce
  API version:	1.37 (minimum version 1.12)
  Go version:	go1.9.4
  Git commit:	0520e24
  Built:	Wed Mar 21 23:08:31 2018
  OS/Arch:	linux/amd64
  Experimental:	false
```

Is this expected behaviour one should work with, or is it a bug? For now, I can work around it by streaming both logs.
It'd be good if one of the suggestions in #1952  were implemented, as I suppose that would make this a non-issue?

**Comment by shin-** [comment]
This is the intended behavior ; `stream=False` returns a binary string containing all logs for the container once they've been retrieved. `stream=True` returns a generator that iterates over log data as it is received. As a result, the correct way to use `logs` with `stream=False` is thus:

```python
err_logs = container.logs(stdout=False, stderr=True, stream=False)
print(err_logs.decode('utf-8'))
```

HTH!
